alchemist-physics/src/main/kotlin/it/unibo/alchemist/model/implementations/properties/PhysicalPedestrian2D.kt

```diff
@@ -63,33 +93,89 @@ class PhysicalPedestrian2D<T>(
+                    assert(it.z == 0.0) { "The cross product result should live on the 2D-plane" }
+                    Euclidean2DPosition(it.x, it.y)
+                }
+            val weightFactor = avoidanceDistanceWeight(other) * avoidanceDirectionWeight(
+                environment.getVelocity(node),
+                environment.getVelocity(other),
+            )
+            if (tangentialForce.magnitude > 0) {
+                return tangentialForce.normalized() * weightFactor
+            }
+        }
+        return Euclidean2DPosition.zero
+    }
+
+    private fun avoidanceDistanceWeight(other: Node<T>): Double {
+        val weight = environment.getDistanceBetweenNodes(node, other) - (rectangleOfInfluenceDimensions.first / 2.0)
+        return weight * weight
+    }
+
+    private fun avoidanceDirectionWeight(thisVelocity: Euclidean2DPosition, otherVelocity: Euclidean2DPosition) =
+        when {
+            thisVelocity.dot(otherVelocity) > 0 -> directionWeight
+            else -> directionWeight * 2
+        }
+
+    override fun fallenAgentAvoidanceForces() =
+        collectForces(::avoid, fallenAgentPerceptionArea) {
+            it.asProperty<T, PhysicalPedestrian2D<T>>().isFallen
+        }
+
+    private fun collectForces(
+        force: (node: Node<T>) -> Euclidean2DPosition,
+        influenceArea: Euclidean2DShape,
+        nodeFilter: (node: Node<T>) -> Boolean = { true },
+    ) = environment.getNodesWithin(influenceArea)
         .asSequence()
         .minusElement(node)
         .filter { it.asPropertyOrNull<T, AreaProperty<T>>() != null }
-        .map { repulsionForce(it) }
-        /*
-         * Discard infinitesimal forces.
-         */
+        .filter(nodeFilter)
+        .map(force)
         .filter { it.magnitude > Double.MIN_VALUE }
         .toList()
 
+    override fun physicalForces(
+        environment: PhysicsEnvironment<T, Euclidean2DPosition, Euclidean2DTransformation, Euclidean2DShapeFactory>,
+    ): List<Euclidean2DPosition> = avoidanceForces()
+
     override fun cloneOnNewNode(node: Node<T>) = PhysicalPedestrian2D(randomGenerator, environment, node)
 
     companion object {
         /**
          * Minimum value for normal state [comfortRay].
          */
-        private const val minimumSpaceTreshold = 0.1
+        private const val minimumSpaceTreshold = 0.5
         /**
          * Maximum value for normal state [comfortRay].
          */
         private const val maximumSpaceThreshold = 1.0
+        /**
+         * Dimension of the rectangle of influence (width, height).
+         */
+        private val rectangleOfInfluenceDimensions: Pair<Double, Double> = Pair(3.0, 1.0)
+        /**
+         * Direction tangential force weight factor for when two nodes are moving in the same direction.
+         */
+        private const val directionWeight = 1.2
+        /**
+         * Fallen agent perception radius.
+         */
+        private const val fallenAgentPerceptionRadius = 1.5
```

Comment:
Where are these parameters from? And what is their unit?

Reply:
I got them from [Pelechano et al](https://repository.upenn.edu/cgi/viewcontent.cgi?article=1223&context=hms) paper. In the paper, they're considered in meters.

Reply:
Then write both the source and the unit in the kdoc, please :)